client/allocrunner/network_manager_linux.go

```diff
@@ -184,7 +184,7 @@ func newNetworkConfigurator(log hclog.Logger, alloc *structs.Allocation, config
 
 	switch {
 	case netMode == "bridge":
-		c, err := newBridgeNetworkConfigurator(log, config.BridgeNetworkName, config.BridgeNetworkAllocSubnet, config.CNIPath, ignorePortMappingHostIP)
+		c, err := newBridgeNetworkConfigurator(log, config.BridgeNetworkName, config.BridgeNetworkAllocSubnet, config.BridgeNetworkHairpinMode, config.CNIPath, ignorePortMappingHostIP)
```

Comment:
The list of arguments is getting uncomfortably long, I wouldn't oppose a new config struct the be passed here. Although `bridgeNetworkConfiguratorConfig` is kind of weird 😬 

Another option would be to pass `config`, but that has the downside of not making it clear which config values are relevant.